vmdb/app/models/ems_microsoft/powershell.rb

```diff
@@ -38,6 +38,10 @@ def log_dos_error_results(results)
         log_header = "MIQ(#{self.class.name}##{__method__})"
         error = results[:data].collect { |d| d[:stderr] }.join
         $scvmm_log.error("#{log_header} #{error}") unless error.blank?
+
+        if error.downcase.include? "'get-scvmmserver' is not recognized"
+          raise "Hyper-V Server is not directly supported. "
+        end
```

Comment:
Maybe have something like the following in the callers:

```ruby
  raise "Error message" unless hyperv_supported?(results)
```

Then, pass the results to the logging method after the conditional `raise`.